homeassistant/components/accuweather/sensor.py

```diff
@@ -71,88 +64,102 @@ class AccuWeatherSensor(CoordinatorEntity, SensorEntity):
     def __init__(
         self,
         name: str,
-        kind: str,
         coordinator: AccuWeatherDataUpdateCoordinator,
+        description: AccuWeatherSensorDescription,
         forecast_day: int | None = None,
     ) -> None:
         """Initialize."""
         super().__init__(coordinator)
-        self._sensor_data = _get_sensor_data(coordinator.data, forecast_day, kind)
-        if forecast_day is None:
-            self._description = SENSOR_TYPES[kind]
-        else:
-            self._description = FORECAST_SENSOR_TYPES[kind]
-        self._unit_system = API_METRIC if coordinator.is_metric else API_IMPERIAL
-        self.kind = kind
+        self._sensor_data = _get_sensor_data(
+            coordinator.data, forecast_day, description.key
+        )
         self._attrs = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        self.forecast_day = forecast_day
-        self._attr_state_class = self._description.get(ATTR_STATE_CLASS)
-        self._attr_icon = self._description[ATTR_ICON]
-        self._attr_device_class = self._description[ATTR_DEVICE_CLASS]
-        self._attr_entity_registry_enabled_default = self._description[ATTR_ENABLED]
-        if self.forecast_day is not None:
-            self._attr_name = f"{name} {self._description[ATTR_LABEL]} {forecast_day}d"
+        if forecast_day is not None:
+            self._attr_name = f"{name} {description.name} {forecast_day}d"
             self._attr_unique_id = (
-                f"{coordinator.location_key}-{kind}-{forecast_day}".lower()
+                f"{coordinator.location_key}-{description.key}-{forecast_day}".lower()
             )
         else:
-            self._attr_name = f"{name} {self._description[ATTR_LABEL]}"
-            self._attr_unique_id = f"{coordinator.location_key}-{kind}".lower()
+            self._attr_name = f"{name} {description.name}"
+            self._attr_unique_id = (
+                f"{coordinator.location_key}-{description.key}".lower()
+            )
         if coordinator.is_metric:
-            self._attr_unit_of_measurement = self._description[ATTR_UNIT_METRIC]
+            self._unit_system = API_METRIC
+            self._attr_unit_of_measurement = description.unit_metric
         else:
-            self._attr_unit_of_measurement = self._description[ATTR_UNIT_IMPERIAL]
+            self._unit_system = API_IMPERIAL
+            self._attr_unit_of_measurement = description.unit_imperial
         self._attr_device_info = {
             "identifiers": {(DOMAIN, coordinator.location_key)},
             "name": NAME,
             "manufacturer": MANUFACTURER,
             "entry_type": "service",
         }
+        self.forecast_day = forecast_day
+        self.entity_description = description
```

Comment:
I usually assign it, directly after the `super().__init__()` call. That way  I'm not going to miss it.